### PR TITLE
[BugFix] Make truncated table not to be moved to trash

### PR DESCRIPTION
**Pull request by zaorangyang**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Currently truncate operation moves the data to the trash directory, which does not match the document description. This pr solves this issue.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3
  - [x] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 11 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/leader/ReportHandler.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|6|6|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java [hunk 1]
```diff
@@ -75,6 +76,8 @@ public class TabletInvertedIndex {
     // replica id -> tablet id
     private Map<Long, Long> replicaToTabletMap = Maps.newHashMap();
 
+    private Set<Long> truncatedTablets = Sets.newHashSet();
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java:79]: shall we persist this meta data? what if FE restarted?

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 2]
```diff
@@ -4521,6 +4521,7 @@ private void truncateTableInternal(OlapTable olapTable, List<Partition> newParti
         // remove the tablets in old partitions
         for (Long tabletId : oldTabletIds) {
             GlobalStateMgr.getCurrentInvertedIndex().deleteTablet(tabletId);
+            GlobalStateMgr.getCurrentInvertedIndex().markTabletTruncated(tabletId);
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:4524]: This mark operation should only be executed in leader node.

**Review by gengjun-git — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**